src/server/shell/accessibility_manager.cpp

```diff
@@ -118,8 +118,24 @@ mir::shell::AccessibilityManager::AccessibilityManager(
     enable_key_repeat{options->get<bool>(options::enable_key_repeat_opt)},
     enable_mouse_keys{options->get<bool>(options::enable_mouse_keys_opt)},
     event_transformer{event_transformer},
-    transformer{std::make_shared<MouseKeysTransformer>()}
+    transformer{[&]
+                {
+                    if (enable_mouse_keys)
+                    {
+                        auto const t = std::make_shared<MouseKeysTransformer>();
+                        event_transformer->append(transformer);
+                        return t;
+                    }
+
+                    return std::shared_ptr<MouseKeysTransformer>{};
+                }()
+
+    }
+{
+}
```

Comment:
I don't think `enable_mouse_keys` is needed anymore

```suggestion
    transformer{[&]
                {
                    return options->get<bool>(options::enable_mouse_keys_opt) ?
                        std::make_shared<MouseKeysTransformer>() :
                        std::shared_ptr<MouseKeysTransformer>{};
                }()
    }
{
    if(transformer)
        event_transformer->append(transformer);
}
```